src/UriParse.c

```diff
@@ -1586,16 +1602,17 @@ static const URI_CHAR * URI_FUNC(ParseSegmentNzNcOrScheme2)(URI_TYPE(ParserState
     case _UT('-'):
     case URI_SET_ALPHA(_UT):
     case URI_SET_DIGIT(_UT):
-        return URI_FUNC(ParseSegmentNzNcOrScheme2)(state, first + 1, afterLast, memory);
+        first += 1;
+        goto tail_call;
 
     case _UT('%'): {
         const URI_CHAR * const afterPctEncoded =
             URI_FUNC(ParsePctEncoded)(state, first, afterLast, memory);
         if (afterPctEncoded == NULL) {
             return NULL;
         }
-        return URI_FUNC(ParseMustBeSegmentNzNc)(state, afterPctEncoded, afterLast,
-                                                memory);
+        first = afterPctEncoded;
+        goto tail_call;
```

Comment:
This one is not a tail call replacement.

Previous: ParseMustBeSegmentNzNc
Now:      ParseSegmentNzNcOrScheme2

Reply:
@TimWolla I confirm, thanks for this crucial finding! Fixed with the next push. I have verified manually myself now that all tail call adjustments match with the contained function's name.